src/main/java/com/example/demo/controller/SurveyController.java

```diff
@@ -50,5 +52,16 @@ public Survey closeSurvey(@PathVariable int id) {
 		survey.setClosed(true);
 		return repository.save(survey);
 	}
+
+	@PostMapping("/survey/{id}/publish")
+	public ResponseEntity<Survey> publishSurvey(@PathVariable int id) {
+		Survey survey = repository.findById(id).get();
+
+		if (survey.isClosed()) {
```

Comment:
Should there extra check to see if survey is already published?

Reply:
Not super critical, publishing a published survey doesn't break anything